### ISSUE TITLE
0.5.0: Python error

### ISSUE THREAD

**Issue by ChristianWeyer** [tillsteinbach/VWsFriend#42]
Hi @tillsteinbach :-)

just tried the latest version 0.5.0:
`docker-compose -f docker-compose-homekit-host.yml --env-file ./myconfig.env up
`
And got the following error:

```
vwsfriend_1   | INFO:VWsFriend:vwsfriend 0.5.0 (using WeConnect-python 0.20.8)
vwsfriend_1   | INFO:weconnect:Could not use token from file /tmp/weconnect.token ([Errno 2] No such file or directory: '/tmp/weconnect.token')
...
vwsfriend_1   | INFO:weconnect:Token refreshed
vwsfriend_1   | INFO:VWsFriend:Starting database upgrade if necessary
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/bin/vwsfriend", line 8, in <module>
vwsfriend_1   |     sys.exit(main())
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 142, in main
vwsfriend_1   |     connector = AgentConnector(weConnect=weConnect, dbUrl=args.dbUrl, interval=args.interval, withDB=args.withDatabase, withABRP=args.withABRP,
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/agent_connector.py", line 32, in __init__
vwsfriend_1   |     run_database_migrations(dsn=dbUrl)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/migrations.py", line 18, in run_database_migrations
vwsfriend_1   |     upgrade(config, "head")
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/command.py", line 320, in upgrade
vwsfriend_1   |     script.run_env()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/script/base.py", line 563, in run_env
vwsfriend_1   |     util.load_python_file(self.dir, "env.py")
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 92, in load_python_file
vwsfriend_1   |     module = load_module_py(module_id, path)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 108, in load_module_py
vwsfriend_1   |     spec.loader.exec_module(module)  # type: ignore
vwsfriend_1   |   File "<frozen importlib._bootstrap_external>", line 855, in exec_module
vwsfriend_1   |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 76, in <module>
vwsfriend_1   |     run_migrations_online()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 58, in run_migrations_online
vwsfriend_1   |     connectable = engine_from_config(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 743, in engine_from_config
vwsfriend_1   |     return create_engine(url, **options)
vwsfriend_1   |   File "<string>", line 2, in create_engine
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py", line 298, in warned
vwsfriend_1   |     return fn(*args, **kwargs)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 548, in create_engine
vwsfriend_1   |     dbapi = dialect_cls.dbapi(**dbapi_args)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 793, in dbapi
vwsfriend_1   |     import psycopg2
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/psycopg2/__init__.py", line 51, in <module>
vwsfriend_1   |     from psycopg2._psycopg import (                     # noqa
vwsfriend_1   | ImportError: libpq.so.5: cannot open shared object file: No such file or directory
vwsfriend_vwsfriend_1 exited with code 1

```

Any clue what could be wrong?
Thanks!

**Comment by tillsteinbach** [comment]
Hi @ChristianWeyer,

doesn't look good :(

Two questions:
- was it working before 0.5.0?
- what is your Docker Host architecture? Some people have trouble on ARM (e.g. Raspberry Pi)

**Comment by ChristianWeyer** [comment]
This is a new Apple MBP with M1 chip - so yeah... could be an issue :-/

**Comment by tillsteinbach** [comment]
Hmm, I,m asking myself if I miss to copy some libraries from the build to the runner stage, but then it should not work at all.

**Comment by ChristianWeyer** [comment]
... happy to try out some things - if you have a clue ... ;-)

**Comment by tillsteinbach** [comment]
I will try to put up an experimental branch and add the library there. Might take some days until I find the time.

If you want to try things out you could login on that image and install libpq-dev, would be a good hint if that works.

**Comment by tillsteinbach** [comment]
Ok, lets go :)

Can you try to replace:
`image: tillsteinbach/vwsfriend:latest`
with
`image: tillsteinbach/vwsfriend:experimental`
and give it another try?

**Comment by ChristianWeyer** [comment]
Some days... jajajaja :-)
Very cool, thanks!

OK, here we go:

```
vwsfriend_1   | INFO:weconnect:Token refreshed
vwsfriend_1   | INFO:VWsFriend:Starting database upgrade if necessary
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3212, in _wrap_pool_connect
vwsfriend_1   |     return fn()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 307, in connect
vwsfriend_1   |     return _ConnectionFairy._checkout(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 767, in _checkout
vwsfriend_1   |     fairy = _ConnectionRecord.checkout(pool)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 425, in checkout
vwsfriend_1   |     rec = pool._do_get()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
vwsfriend_1   |     return self._create_connection()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 253, in _create_connection
vwsfriend_1   |     return _ConnectionRecord(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 368, in __init__
vwsfriend_1   |     self.__connect()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 611, in __connect
vwsfriend_1   |     pool.logger.debug("Error on connect(): %s", e)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
vwsfriend_1   |     compat.raise_(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1   |     raise exception
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 605, in __connect
vwsfriend_1   |     connection = pool._invoke_creator(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 578, in connect
vwsfriend_1   |     return dialect.connect(*cargs, **cparams)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 584, in connect
vwsfriend_1   |     return self.dbapi.connect(*cargs, **cparams)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/psycopg2/__init__.py", line 122, in connect
vwsfriend_1   |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
vwsfriend_1   | psycopg2.OperationalError: could not connect to server: Connection refused
vwsfriend_1   | 	Is the server running on host "postgresdbbackend" (127.0.0.1) and accepting
vwsfriend_1   | 	TCP/IP connections on port 5432?
vwsfriend_1   |
vwsfriend_1   |
vwsfriend_1   | The above exception was the direct cause of the following exception:
vwsfriend_1   |
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/bin/vwsfriend", line 8, in <module>
vwsfriend_1   |     sys.exit(main())
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 142, in main
vwsfriend_1   |     connector = AgentConnector(weConnect=weConnect, dbUrl=args.dbUrl, interval=args.interval, withDB=args.withDatabase, withABRP=args.withABRP,
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/agent_connector.py", line 32, in __init__
vwsfriend_1   |     run_database_migrations(dsn=dbUrl)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/migrations.py", line 18, in run_database_migrations
vwsfriend_1   |     upgrade(config, "head")
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/command.py", line 320, in upgrade
vwsfriend_1   |     script.run_env()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/script/base.py", line 563, in run_env
vwsfriend_1   |     util.load_python_file(self.dir, "env.py")
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 92, in load_python_file
vwsfriend_1   |     module = load_module_py(module_id, path)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 108, in load_module_py
vwsfriend_1   |     spec.loader.exec_module(module)  # type: ignore
vwsfriend_1   |   File "<frozen importlib._bootstrap_external>", line 855, in exec_module
vwsfriend_1   |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 76, in <module>
vwsfriend_1   |     run_migrations_online()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 64, in run_migrations_online
vwsfriend_1   |     with connectable.connect() as connection:
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3166, in connect
vwsfriend_1   |     return self._connection_cls(self, close_with_result=close_with_result)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
vwsfriend_1   |     else engine.raw_connection()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3245, in raw_connection
vwsfriend_1   |     return self._wrap_pool_connect(self.pool.connect, _connection)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3215, in _wrap_pool_connect
vwsfriend_1   |     Connection._handle_dbapi_exception_noconnection(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2069, in _handle_dbapi_exception_noconnection
vwsfriend_1   |     util.raise_(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1   |     raise exception
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3212, in _wrap_pool_connect
vwsfriend_1   |     return fn()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 307, in connect
vwsfriend_1   |     return _ConnectionFairy._checkout(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 767, in _checkout
vwsfriend_1   |     fairy = _ConnectionRecord.checkout(pool)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 425, in checkout
vwsfriend_1   |     rec = pool._do_get()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
vwsfriend_1   |     return self._create_connection()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 253, in _create_connection
vwsfriend_1   |     return _ConnectionRecord(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 368, in __init__
vwsfriend_1   |     self.__connect()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 611, in __connect
vwsfriend_1   |     pool.logger.debug("Error on connect(): %s", e)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
vwsfriend_1   |     compat.raise_(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1   |     raise exception
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 605, in __connect
vwsfriend_1   |     connection = pool._invoke_creator(self)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 578, in connect
vwsfriend_1   |     return dialect.connect(*cargs, **cparams)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 584, in connect
vwsfriend_1   |     return self.dbapi.connect(*cargs, **cparams)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/psycopg2/__init__.py", line 122, in connect
vwsfriend_1   |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
vwsfriend_1   | sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not connect to server: Connection refused
vwsfriend_1   | 	Is the server
``` running on host "postgresdbbackend" (127.0.0.1) and accepting
vwsfriend_1   | 	TCP/IP connections on port 5432?
vwsfriend_1   |
vwsfriend_1   | (Background on this error at: https://sqlalche.me/e/14/e3q8)
vwsfriend_vwsfriend_1 exited with code 1

**Comment by tillsteinbach** [comment]
Hmm, is the Postgres server running? Maybe it takes too much time to Startup. Let me think…

**Comment by ChristianWeyer** [comment]
It should run:
```

postgresdb_1  | PostgreSQL init process complete; ready for start up.
postgresdb_1  |
postgresdb_1  | 2021-09-11 12:44:58.863 UTC [1] LOG:  starting PostgreSQL 13.4 (Debian 13.4-1.pgdg100+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
postgresdb_1  | 2021-09-11 12:44:58.863 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgresdb_1  | 2021-09-11 12:44:58.863 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgresdb_1  | 2021-09-11 12:44:58.865 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgresdb_1  | 2021-09-11 12:44:58.868 UTC [62] LOG:  database system was shut down at 2021-09-11 12:44:58 UTC
postgresdb_1  | 2021-09-11 12:44:58.870 UTC [1] LOG:  database system is ready to accept connections
```

**Comment by tillsteinbach** [comment]
What happens if you keep Postgres running and just restart vwsfriend?

**Comment by ChristianWeyer** [comment]
What would be the best way to do this with docker compose?

**Comment by tillsteinbach** [comment]
`docker-compose up -d` will start up the stack but puts the processes in the background, you can then do `docker-compose up vwsfriend` to start just the vwsfriend container

**Comment by ChristianWeyer** [comment]
I used this command:
`docker compose -f docker-compose-homekit-host.yml --env-file ./myconfig.env up vwsfriend`

And still got the exception:

```
vwsfriend_1  | INFO:VWsFriend:vwsfriend develop (using WeConnect-python 0.20.8)
vwsfriend_1  | INFO:weconnect:Could not use token from file /tmp/weconnect.token ([Errno 2] No such file or directory: '/tmp/weconnect.token')
vwsfriend_1  | INFO:weconnect:Token refreshed
vwsfriend_1  | INFO:VWsFriend:Starting database upgrade if necessary
vwsfriend_1  | Traceback (most recent call last):
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3212, in _wrap_pool_connect
vwsfriend_1  |     return fn()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 307, in connect
vwsfriend_1  |     return _ConnectionFairy._checkout(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 767, in _checkout
vwsfriend_1  |     fairy = _ConnectionRecord.checkout(pool)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 425, in checkout
vwsfriend_1  |     rec = pool._do_get()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
vwsfriend_1  |     return self._create_connection()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 253, in _create_connection
vwsfriend_1  |     return _ConnectionRecord(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 368, in __init__
vwsfriend_1  |     self.__connect()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 611, in __connect
vwsfriend_1  |     pool.logger.debug("Error on connect(): %s", e)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
vwsfriend_1  |     compat.raise_(
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1  |     raise exception
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 605, in __connect
vwsfriend_1  |     connection = pool._invoke_creator(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 578, in connect
vwsfriend_1  |     return dialect.connect(*cargs, **cparams)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 584, in connect
vwsfriend_1  |     return self.dbapi.connect(*cargs, **cparams)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/psycopg2/__init__.py", line 122, in connect
vwsfriend_1  |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
vwsfriend_1  | psycopg2.OperationalError: could not connect to server: Connection refused
vwsfriend_1  | 	Is the server running on host "postgresdbbackend" (127.0.0.1) and accepting
vwsfriend_1  | 	TCP/IP connections on port 5432?
vwsfriend_1  |
vwsfriend_1  |
vwsfriend_1  | The above exception was the direct cause of the following exception:
vwsfriend_1  |
vwsfriend_1  | Traceback (most recent call last):
vwsfriend_1  |   File "/opt/venv/bin/vwsfriend", line 8, in <module>
vwsfriend_1  |     sys.exit(main())
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 142, in main
vwsfriend_1  |     connector = AgentConnector(weConnect=weConnect, dbUrl=args.dbUrl, interval=args.interval, withDB=args.withDatabase, withABRP=args.withABRP,
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/agent_connector.py", line 32, in __init__
vwsfriend_1  |     run_database_migrations(dsn=dbUrl)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/migrations.py", line 18, in run_database_migrations
vwsfriend_1  |     upgrade(config, "head")
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/alembic/command.py", line 320, in upgrade
vwsfriend_1  |     script.run_env()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/alembic/script/base.py", line 563, in run_env
vwsfriend_1  |     util.load_python_file(self.dir, "env.py")
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 92, in load_python_file
vwsfriend_1  |     module = load_module_py(module_id, path)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/alembic/util/pyfiles.py", line 108, in load_module_py
vwsfriend_1  |     spec.loader.exec_module(module)  # type: ignore
vwsfriend_1  |   File "<frozen importlib._bootstrap_external>", line 855, in exec_module
vwsfriend_1  |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 76, in <module>
vwsfriend_1  |     run_migrations_online()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/model/vwsfriend-schema/env.py", line 64, in run_migrations_online
vwsfriend_1  |     with connectable.connect() as connection:
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3166, in connect
vwsfriend_1  |     return self._connection_cls(self, close_with_result=close_with_result)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
vwsfriend_1  |     else engine.raw_connection()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3245, in raw_connection
vwsfriend_1  |     return self._wrap_pool_connect(self.pool.connect, _connection)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3215, in _wrap_pool_connect
vwsfriend_1  |     Connection._handle_dbapi_exception_noconnection(
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2069, in _handle_dbapi_exception_noconnection
vwsfriend_1  |     util.raise_(
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1  |     raise exception
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 3212, in _wrap_pool_connect
vwsfriend_1  |     return fn()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 307, in connect
vwsfriend_1  |     return _ConnectionFairy._checkout(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 767, in _checkout
vwsfriend_1  |     fairy = _ConnectionRecord.checkout(pool)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 425, in checkout
vwsfriend_1  |     rec = pool._do_get()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
vwsfriend_1  |     return self._create_connection()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 253, in _create_connection
vwsfriend_1  |     return _ConnectionRecord(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 368, in __init__
vwsfriend_1  |     self.__connect()
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 611, in __connect
vwsfriend_1  |     pool.logger.debug("Error on connect(): %s", e)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
vwsfriend_1  |     compat.raise_(
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
vwsfriend_1  |     raise exception
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 605, in __connect
vwsfriend_1  |     connection = pool._invoke_creator(self)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 578, in connect
vwsfriend_1  |     return dialect.connect(*cargs, **cparams)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 584, in connect
vwsfriend_1  |     return self.dbapi.connect(*cargs, **cparams)
vwsfriend_1  |   File "/opt/venv/lib/python3.9/site-packages/psycopg2/__init__.py", line 122, in connect
vwsfriend_1  |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
vwsfriend_1  | sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not connect to server: Connection refused
vwsfriend_1  | 	Is the server running on host "postgresdbbackend" (127.0.0.1) and accepting
vwsfriend_1  | 	TCP/IP connections on port 5432?
vwsfriend_1  |
vwsfriend_1  | (Background on this error at: https://sqlalche.me/e/14/e3q8)
vwsfriend_1 exited with code 1
```

**Comment by tillsteinbach** [comment]
hmmmm, why can't it reach the database...

**Comment by tillsteinbach** [comment]
I think I found the problem. The homekit-host config is missing `ports:
      - 5432:5432`
Can you use the new one I just uploaded? (remember to change to "experimental" for now until I merged that into the main branch

**Comment by ChristianWeyer** [comment]
Looks better :-)

But when surfing to http://localhost:4000/ I get

![image](https://user-images.githubusercontent.com/888718/132952730-8904de5a-041e-4f79-becc-2920a98805b6.png)

**Comment by ChristianWeyer** [comment]
But it says

```
vwsfriend_1   | INFO:VWsFriend:VWsFriend is listening on 0.0.0.0:4000)
vwsfriend_1   | INFO:VWsFriend:Updating data from WeConnect
```

**Comment by tillsteinbach** [comment]
We will get there ;) I just saw that the Dockerfile may be missing an EXPOSE 4000. I'm just building a new image. Should be there in ~30min

**Comment by tillsteinbach** [comment]
BTW: Thanks for debugging with me. Improved the stability already a lot today!

**Comment by ChristianWeyer** [comment]
Gerne ;-)

**Comment by tillsteinbach** [comment]
Updated Image should be there

**Comment by ChristianWeyer** [comment]
I deleted all the images locally and pulled again.

Still, the same error...

**Comment by tillsteinbach** [comment]
Hmm, will try out some things later

**Comment by tillsteinbach** [comment]
If I understand the internet right docker host mode on OS X behaves a bit different than on other OS. I now also understand why. Docker is running a linux container, thus it needs to run it in a virtual machine. So probably you cannot connect to the container via localhost but some other IP address that you need to find out.
The problem is: The whole purpose of the host networking is to make multicast work (what is necessary for HomeKit). I expect that this won't work in this virtual machine anyways. so the whole purpose of the host network setup is gone.

To sum it up: You can probably only use the config without HomeKit. Or try to find a linux machine to host the docker :(

**Comment by ChristianWeyer** [comment]
... will take some time for me to test this out ... busy :-/

**Comment by tillsteinbach** [comment]
No problem. If you don't mind I close this issue for now.
